add missing internal panic in cairo1

zuphitf opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/starkware-libs/blockifier/1340)
<!-- Reviewable:end -->

ilyalesokhin-starkware reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1340)* status: 0 of 6 files reviewed, 1 unresolved discussion (waiting on @dorimedini-starkware and @zuphitf)

___
*[`crates/blockifier/feature_contracts/cairo0/test_contract.cairo` line 243 at r1](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoGEPAH2RpPf87bxvJZ:-NoGEPAH2RpPf87bxvJ_:bi39py3) ([raw file](https://github.com/starkware-libs/blockifier/blob/edfd949a2a21651399b413a5c24d1a0cbc74346b/crates/blockifier/feature_contracts/cairo0/test_contract.cairo#L243)):*
> ```Cairo
> 
> @external
> func recursive_syscall{syscall_ptr: felt*}(contract_address: felt, function_selector: felt, depth: felt) {
> ```

do you need to change the contract in this PR?



_Code quote:_
```Cairo
func recursive_syscall{syscall_ptr: felt*}(contract_address: felt, function_selector: felt, depth: felt) {
```


<!-- Sent from Reviewable.io -->

dorimedini-starkware reviewed (changes requested):
> Reviewed 6 of 6 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1340)* status: all files reviewed, 5 unresolved discussions (waiting on @zuphitf)

___
*[`crates/blockifier/feature_contracts/cairo0/test_contract.cairo` line 250 at r1](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoGK77w9JAQLwGw9id1:-NoGK77xAkTMYlr9EQhg:b-i9yswe) ([raw file](https://github.com/starkware-libs/blockifier/blob/edfd949a2a21651399b413a5c24d1a0cbc74346b/crates/blockifier/feature_contracts/cairo0/test_contract.cairo#L250)):*
> ```Cairo
>     }
>     if (function_selector == recursive_fail_selector) {
>         // Call 'recursive_fail`.
> ```

backticks *xor* single quotes



_Code quote:_
```Cairo
'recursive_fail`
```
___
*[`crates/blockifier/src/execution/entry_point_execution.rs` line 119 at r1](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoGL2dh8e4Y55eCgNvc:-NoGL2dh8e4Y55eCgNvd:bt7jdj7) ([raw file](https://github.com/starkware-libs/blockifier/blob/edfd949a2a21651399b413a5c24d1a0cbc74346b/crates/blockifier/src/execution/entry_point_execution.rs#L119)):*
> ```XML
>             address,
>             format!("Panicked with: {}", format_panic_data(&call_info.execution.retdata.0)),
>         ));
> ```

after this change, can you parametrize `test_stack_trace` to also use the Cairo1 contract?



_Code quote:_
```XML
        context.error_stack.push((
            address,
            format!("Panicked with: {}", format_panic_data(&call_info.execution.retdata.0)),
        ));
```
___
*[`crates/blockifier/src/transaction/account_transactions_test.rs` line 1118 at r1](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoGKaW4BvX_W5MCDjj1:-NoGKaW4BvX_W5MCDjj2:b6k10yq) ([raw file](https://github.com/starkware-libs/blockifier/blob/edfd949a2a21651399b413a5c24d1a0cbc74346b/crates/blockifier/src/transaction/account_transactions_test.rs#L1118)):*
> ```XML
> 
> #[rstest]
> /// Tests that error types are as expected.
> ```

where is there a type check? I only see `is_some()`



_Code quote:_
```XML
/// Tests that error types are as expected.
```
___
*[`crates/blockifier/src/transaction/account_transactions_test.rs` line 1142 at r1](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoGKWr21Xz5hZqCOmWB:-NoGKWr34lF8koDeaF5_:b-n13iyj) ([raw file](https://github.com/starkware-libs/blockifier/blob/edfd949a2a21651399b413a5c24d1a0cbc74346b/crates/blockifier/src/transaction/account_transactions_test.rs#L1142)):*
> ```XML
>     };
>     let tx_execution_info = run_invoke_tx(&mut state, &block_context, invoke_args.clone());
>     assert!(tx_execution_info.unwrap().revert_error.is_some());
> ```

are you sure this test failed before your change?



_Code quote:_
```XML
assert!(tx_execution_info.unwrap().revert_error.is_some());
```


<!-- Sent from Reviewable.io -->

dorimedini-starkware reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1340)* status: all files reviewed, 5 unresolved discussions (waiting on @ilyalesokhin-starkware and @zuphitf)

___
*[`crates/blockifier/feature_contracts/cairo0/test_contract.cairo` line 243 at r1](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoGEPAH2RpPf87bxvJZ:-NoGZ0DSBgvrKcI4xDNc:bbw19bq) ([raw file](https://github.com/starkware-libs/blockifier/blob/edfd949a2a21651399b413a5c24d1a0cbc74346b/crates/blockifier/feature_contracts/cairo0/test_contract.cairo#L243)):*
<details><summary><i>Previously, ilyalesokhin-starkware wrote…</i></summary><blockquote>

do you need to change the contract in this PR?
</blockquote></details>

if the cairo1 contract needs to change I prefer the cairo0 contract to be in the same PR


<!-- Sent from Reviewable.io -->

zuphitf reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1340)* status: 5 of 6 files reviewed, 4 unresolved discussions (waiting on @dorimedini-starkware and @ilyalesokhin-starkware)

___
*[`crates/blockifier/feature_contracts/cairo0/test_contract.cairo` line 243 at r1](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoGEPAH2RpPf87bxvJZ:-NoGhEJ6CuHFUY3-vigX:b-gofqmk) ([raw file](https://github.com/starkware-libs/blockifier/blob/edfd949a2a21651399b413a5c24d1a0cbc74346b/crates/blockifier/feature_contracts/cairo0/test_contract.cairo#L243)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

if the cairo1 contract needs to change I prefer the cairo0 contract to be in the same PR
</blockquote></details>

I order to create this specific output yeah. But I guess this could be split if you really wanna.
___
*[`crates/blockifier/feature_contracts/cairo0/test_contract.cairo` line 250 at r1](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoGK77w9JAQLwGw9id1:-NoGjGJX52wwy5uUFjC-:b7ubq0d) ([raw file](https://github.com/starkware-libs/blockifier/blob/edfd949a2a21651399b413a5c24d1a0cbc74346b/crates/blockifier/feature_contracts/cairo0/test_contract.cairo#L250)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

backticks *xor* single quotes
</blockquote></details>

sounds like a good practice
___
*[`crates/blockifier/src/execution/entry_point_execution.rs` line 119 at r1](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoGL2dh8e4Y55eCgNvc:-NoGixR51i2q6xFJj33p:b28z2i4) ([raw file](https://github.com/starkware-libs/blockifier/blob/edfd949a2a21651399b413a5c24d1a0cbc74346b/crates/blockifier/src/execution/entry_point_execution.rs#L119)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

after this change, can you parametrize `test_stack_trace` to also use the Cairo1 contract?
</blockquote></details>

In this PR or a separate one?
___
*[`crates/blockifier/src/transaction/account_transactions_test.rs` line 1118 at r1](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoGKaW4BvX_W5MCDjj1:-NoGiEp0EOVo6H2kt9aV:b-1xxfov) ([raw file](https://github.com/starkware-libs/blockifier/blob/edfd949a2a21651399b413a5c24d1a0cbc74346b/crates/blockifier/src/transaction/account_transactions_test.rs#L1118)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

where is there a type check? I only see `is_some()`
</blockquote></details>

This is not supposed to be the final form of this test. It's just a placeholder with the general flow for now. That's also why the assert is trivial. It was a pretty good base for debugging (added the output in the description).
I can remove it for now and add it back later with more content, or leave it in so it's easier to checkout and alter in future PRs.


<!-- Sent from Reviewable.io -->

dorimedini-starkware reviewed (changes requested):
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1340)* status: all files reviewed, 3 unresolved discussions (waiting on @ilyalesokhin-starkware and @zuphitf)

___
*[`crates/blockifier/src/execution/entry_point_execution.rs` line 119 at r1](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoGL2dh8e4Y55eCgNvc:-NoGouCx5_p3l0e1JxYu:bjhb4az) ([raw file](https://github.com/starkware-libs/blockifier/blob/edfd949a2a21651399b413a5c24d1a0cbc74346b/crates/blockifier/src/execution/entry_point_execution.rs#L119)):*
<details><summary><i>Previously, zuphitf wrote…</i></summary><blockquote>

In this PR or a separate one?
</blockquote></details>

I think that if you parametrize `test_stack_trace` (with different expected regression strings...) in this PR, you could delete `test_error_stack1`. Right?
If so, please do


<!-- Sent from Reviewable.io -->

zuphitf commented:
*[`crates/blockifier/src/execution/entry_point_execution.rs` line 119 at r1](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoGL2dh8e4Y55eCgNvc:-NoIlQlxAsstA6Y3MV1h:b35s5d9) ([raw file](https://github.com/starkware-libs/blockifier/blob/edfd949a2a21651399b413a5c24d1a0cbc74346b/crates/blockifier/src/execution/entry_point_execution.rs#L119)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

I think that if you parametrize `test_stack_trace` (with different expected regression strings...) in this PR, you could delete `test_error_stack1`. Right?
If so, please do
</blockquote></details>

I had split it out. Updated now. The other part will be in a separate PR.


<!-- Sent from Reviewable.io -->

dorimedini-starkware reviewed (changes requested):
> Reviewed 8 of 8 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1340)* status: all files reviewed, 7 unresolved discussions (waiting on @ilyalesokhin-starkware and @zuphitf)

___
*[`crates/blockifier/src/test_utils.rs` line 206 at r3](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoIwm8E1eMXpJXYqvr0:-NoIwm8E1eMXpJXYqvr1:b337hvh) ([raw file](https://github.com/starkware-libs/blockifier/blob/557011b31c98e584c37f4ede4c2271dd22e809da/crates/blockifier/src/test_utils.rs#L206)):*
> ```XML
> ) -> CallEntryPoint {
>     let default_contract_address = contract_address!(TEST_CONTRACT_ADDRESS);
>     let contract_address = contract_address.unwrap_or(default_contract_address);
> ```

no need for `Option` IMO



_Suggestion:_
```XML
pub fn trivial_external_entry_point() -> CallEntryPoint {
    // TODO(Dori, 1/5/2024): Stop using `TEST_CONTRACT_ADDRESS`.
    trivial_external_entry_point_with_address(contract_address!(TEST_CONTRACT_ADDRESS))
}

pub fn trivial_external_entry_point_with_address(
    contract_address: ContractAddress
) -> CallEntryPoint {
```
___
*[`crates/blockifier/src/execution/entry_point_execution.rs` line 118 at r3](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoIx37V7yNStOROACra:-NoIx37V7yNStOROACrb:brz8uzt) ([raw file](https://github.com/starkware-libs/blockifier/blob/557011b31c98e584c37f4ede4c2271dd22e809da/crates/blockifier/src/execution/entry_point_execution.rs#L118)):*
> ```XML
>         context.error_stack.push((
>             address,
>             format!("Panicked with: {}\n", format_panic_data(&call_info.execution.retdata.0)),
> ```

why?
if this is for stack trace formatting, please remove it and add it in the function that constructs the pretty output



_Code quote:_
```XML
\n
```
___
*[`crates/blockifier/src/execution/entry_point_test.rs` line 592 at r3](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoIxM1IEi761gT0Bfn4:-NoIxM1IEi761gT0Bfn5:bne6ags) ([raw file](https://github.com/starkware-libs/blockifier/blob/557011b31c98e584c37f4ede4c2271dd22e809da/crates/blockifier/src/execution/entry_point_test.rs#L592)):*
> ```XML
>                 .offset
>         }
>     };
> ```

would this work?



_Suggestion:_
```XML
    let entry_point_offset = (match contract_class {
        ContractClass::V0(class) => {
            class
                .entry_points_by_type
                .get(&EntryPointType::External)
        }
        ContractClass::V1(class) => {
            class
                .entry_points_by_type
                .get(&EntryPointType::External)
        }
    })
    .unwrap()
    .iter()
    .find(|ep| ep.selector == entry_point_call.entry_point_selector)
    .unwrap()
    .offset;
```
___
*[`crates/blockifier/src/execution/entry_point_test.rs` line 595 at r3](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoIxkwg-wKGEmGgQ3Cq:-NoIxkwg-wKGEmGgQ3Cr:bbyxko4) ([raw file](https://github.com/starkware-libs/blockifier/blob/557011b31c98e584c37f4ede4c2271dd22e809da/crates/blockifier/src/execution/entry_point_test.rs#L595)):*
> ```XML
>     // Relative offsets of the test_call_contract entry point and the inner call.
>     let call_location = entry_point_offset.0 + 14;
>     let entry_point_location = entry_point_offset.0 - 3;
> ```

can you use these values in the cairo1 trace output?



_Code quote:_
```XML
    let call_location = entry_point_offset.0 + 14;
    let entry_point_location = entry_point_offset.0 - 3;
```


<!-- Sent from Reviewable.io -->

zuphitf reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1340)* status: 5 of 8 files reviewed, 7 unresolved discussions (waiting on @dorimedini-starkware and @ilyalesokhin-starkware)

___
*[`crates/blockifier/src/test_utils.rs` line 206 at r3](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoIwm8E1eMXpJXYqvr0:-NoMFlge5R6D9iEP_eku:b-896fix) ([raw file](https://github.com/starkware-libs/blockifier/blob/557011b31c98e584c37f4ede4c2271dd22e809da/crates/blockifier/src/test_utils.rs#L206)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

no need for `Option` IMO
</blockquote></details>

Done.
___
*[`crates/blockifier/src/execution/entry_point_execution.rs` line 118 at r1](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoGF6YB6TqWgLJZQayS:-NoMa7587VA6mVu27NEr:b-ey6ff6) ([raw file](https://github.com/starkware-libs/blockifier/blob/edfd949a2a21651399b413a5c24d1a0cbc74346b/crates/blockifier/src/execution/entry_point_execution.rs#L118)):*
> ```XML
>         context.error_stack.push((
>             address,
>             format!("Panicked with: {}", format_panic_data(&call_info.execution.retdata.0)),
> ```

Done through `ExecutionFailed` formatting
___
*[`crates/blockifier/src/execution/entry_point_execution.rs` line 122 at r1](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoIZ-1_A2dMZhiUOamM:-NoMaFpN8HIDZyJb1Fns:b-kpb6f0) ([raw file](https://github.com/starkware-libs/blockifier/blob/edfd949a2a21651399b413a5c24d1a0cbc74346b/crates/blockifier/src/execution/entry_point_execution.rs#L122)):*
<details><summary><i>Previously, ilyalesokhin-starkware wrote…</i></summary><blockquote>

I don't like that we print the same error twice

Maybe something like:
</blockquote></details>

Done on the first half.
`InnerCallFailed` doesn't exist. 
___
*[`crates/blockifier/src/execution/entry_point_execution.rs` line 118 at r3](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoIx37V7yNStOROACra:-NoMbN-J7PbAvfSlJRc1:b-lzycl7) ([raw file](https://github.com/starkware-libs/blockifier/blob/557011b31c98e584c37f4ede4c2271dd22e809da/crates/blockifier/src/execution/entry_point_execution.rs#L118)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

why?
if this is for stack trace formatting, please remove it and add it in the function that constructs the pretty output
</blockquote></details>

This should be here, like we discussed
___
*[`crates/blockifier/src/execution/entry_point_test.rs` line 592 at r3](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoIxM1IEi761gT0Bfn4:-NoLmnfY4TsoRmeoRdu5:bb710yj) ([raw file](https://github.com/starkware-libs/blockifier/blob/557011b31c98e584c37f4ede4c2271dd22e809da/crates/blockifier/src/execution/entry_point_test.rs#L592)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

would this work?
</blockquote></details>

Nope
___
*[`crates/blockifier/src/execution/entry_point_test.rs` line 595 at r3](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoIxkwg-wKGEmGgQ3Cq:-NoMwXlN6XRPwxKxkBka:b4nsjg5) ([raw file](https://github.com/starkware-libs/blockifier/blob/557011b31c98e584c37f4ede4c2271dd22e809da/crates/blockifier/src/execution/entry_point_test.rs#L595)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

can you use these values in the cairo1 trace output?
</blockquote></details>

next time


<!-- Sent from Reviewable.io -->

dorimedini-starkware reviewed (dismissed):
> Reviewed 3 of 3 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1340)* status: all files reviewed, 4 unresolved discussions (waiting on @ilyalesokhin-starkware and @zuphitf)

___
*[`crates/blockifier/src/execution/entry_point_execution.rs` line 119 at r4](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoMxdq21oILdNEZOgu0:-NoMxdq21oILdNEZOgu1:b-q9pgqq) ([raw file](https://github.com/starkware-libs/blockifier/blob/4ae196a388a38b275f6a8e9ba57b605e05661cb7/crates/blockifier/src/execution/entry_point_execution.rs#L119)):*
> ```XML
>             EntryPointExecutionError::ExecutionFailed { error_data: error_data.clone() };
>         context.error_stack.push((address, format!("{}\n", exec_error)));
>         return Err(EntryPointExecutionError::ExecutionFailed { error_data });
> ```

is `EntryPointExecutionError` `clone`able?
if so, please make this change (non-blocking)



_Suggestion:_
```XML
        let exec_error = EntryPointExecutionError::ExecutionFailed { error_data };
        context.error_stack.push((address, format!("{}\n", exec_error.clone())));
        return Err(exec_error);
```


<!-- Sent from Reviewable.io -->

dorimedini-starkware reviewed (approved):
> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1340)* status: all files reviewed, 3 unresolved discussions (waiting on @ilyalesokhin-starkware)



<!-- Sent from Reviewable.io -->

ilyalesokhin-starkware commented:
*[`crates/blockifier/src/execution/entry_point_test.rs` line 633 at r5](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoN2JD95eOzGCrLJdku:-NoN2JD95eOzGCrLJdkv:b-c03x8j) ([raw file](https://github.com/starkware-libs/blockifier/blob/a770b76e9a8491012e57c31f532216e3e009226f/crates/blockifier/src/execution/entry_point_test.rs#L633)):*
> ```XML
> Error at pc=0:4798:
> Got an exception while executing a hint: Hint Error: Execution failed. Failure reason: 0x6661696c \
>          ('fail').
> ```

I want something like.



_Suggestion:_
```XML
Got an exception while executing a hint: Hint Error: call_contract syscall failed.
```


<!-- Sent from Reviewable.io -->

ilyalesokhin-starkware reviewed (commented):
> Reviewed 2 of 8 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1340)* status: all files reviewed, 3 unresolved discussions (waiting on @ArielElp and @zuphitf)

___
*[`crates/blockifier/src/execution/entry_point_execution.rs` line 119 at r1](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoGL2dh8e4Y55eCgNvc:-NoN29FEC6TBH2FLqw3d:bjz5t77) ([raw file](https://github.com/starkware-libs/blockifier/blob/edfd949a2a21651399b413a5c24d1a0cbc74346b/crates/blockifier/src/execution/entry_point_execution.rs#L119)):*
<details><summary><i>Previously, zuphitf wrote…</i></summary><blockquote>

I had split it out. Updated now. The other part will be in a separate PR.
</blockquote></details>

I still see contract changes in this PR


<!-- Sent from Reviewable.io -->

zuphitf commented:
*[`crates/blockifier/src/execution/entry_point_execution.rs` line 119 at r1](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoGL2dh8e4Y55eCgNvc:-NoQOT5x59tvILW82S_K:b-rirq3q) ([raw file](https://github.com/starkware-libs/blockifier/blob/edfd949a2a21651399b413a5c24d1a0cbc74346b/crates/blockifier/src/execution/entry_point_execution.rs#L119)):*
<details><summary><i>Previously, ilyalesokhin-starkware wrote…</i></summary><blockquote>

I still see contract changes in this PR
</blockquote></details>

Yes, those are needed for the trace regression test


<!-- Sent from Reviewable.io -->

dorimedini-starkware reviewed (dismissed):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1340)* status: all files reviewed, 3 unresolved discussions (waiting on @ArielElp, @ilyalesokhin-starkware, and @zuphitf)

___
*[`crates/blockifier/src/execution/entry_point_execution.rs` line 119 at r1](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoGL2dh8e4Y55eCgNvc:-NoQfHub7NWvztIYDG1A:bz02yub) ([raw file](https://github.com/starkware-libs/blockifier/blob/edfd949a2a21651399b413a5c24d1a0cbc74346b/crates/blockifier/src/execution/entry_point_execution.rs#L119)):*
<details><summary><i>Previously, zuphitf wrote…</i></summary><blockquote>

Yes, those are needed for the trace regression test
</blockquote></details>

@ilyalesokhin-starkware these are test contracts. changes in tests contracts relevant to the tested feature are fine in the same PR IMO


<!-- Sent from Reviewable.io -->

ilyalesokhin-starkware reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1340)* status: all files reviewed, 3 unresolved discussions (waiting on @ArielElp and @zuphitf)

___
*[`crates/blockifier/src/execution/entry_point_execution.rs` line 122 at r1](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoIZ-1_A2dMZhiUOamM:-NoRmeicBnvzgmePQopr:b-t95nzl) ([raw file](https://github.com/starkware-libs/blockifier/blob/edfd949a2a21651399b413a5c24d1a0cbc74346b/crates/blockifier/src/execution/entry_point_execution.rs#L122)):*
<details><summary><i>Previously, zuphitf wrote…</i></summary><blockquote>

Done on the first half.
`InnerCallFailed` doesn't exist.
</blockquote></details>

add it


<!-- Sent from Reviewable.io -->

ilyalesokhin-starkware commented:
*[`crates/blockifier/src/execution/entry_point_execution.rs` line 119 at r1](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoGL2dh8e4Y55eCgNvc:-NoRoDA87ezHWipG_Tu-:b-vpjfe1) ([raw file](https://github.com/starkware-libs/blockifier/blob/edfd949a2a21651399b413a5c24d1a0cbc74346b/crates/blockifier/src/execution/entry_point_execution.rs#L119)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

@ilyalesokhin-starkware these are test contracts. changes in tests contracts relevant to the tested feature are fine in the same PR IMO
</blockquote></details>

By doing them in the same PR you make it hard to see the change.


<!-- Sent from Reviewable.io -->

zuphitf reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1340)* status: 7 of 9 files reviewed, 3 unresolved discussions (waiting on @ArielElp, @dorimedini-starkware, and @ilyalesokhin-starkware)

___
*[`crates/blockifier/src/execution/entry_point_execution.rs` line 122 at r1](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoIZ-1_A2dMZhiUOamM:-NofIwz20t1aPvD6PdMr:b-jkpwda) ([raw file](https://github.com/starkware-libs/blockifier/blob/edfd949a2a21651399b413a5c24d1a0cbc74346b/crates/blockifier/src/execution/entry_point_execution.rs#L122)):*
<details><summary><i>Previously, ilyalesokhin-starkware wrote…</i></summary><blockquote>

add it
</blockquote></details>

I don't think it's a good idea.
In fact, I don't think this is the right place to do this.
I'm reverting the changes in this file and adding the missing match in the wrapping function, where it belongs.


<!-- Sent from Reviewable.io -->

zuphitf commented:
*[`crates/blockifier/src/execution/entry_point_test.rs` line 633 at r5](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoN2JD95eOzGCrLJdku:-NofOxdc4HsiiuTtZJNv:ben3qbt) ([raw file](https://github.com/starkware-libs/blockifier/blob/a770b76e9a8491012e57c31f532216e3e009226f/crates/blockifier/src/execution/entry_point_test.rs#L633)):*
<details><summary><i>Previously, ilyalesokhin-starkware wrote…</i></summary><blockquote>

I want something like.
</blockquote></details>

Didn't we discuss with Ariel that this is in the todo / nice to have list for the future refactoring and not a requirement for this PR?


<!-- Sent from Reviewable.io -->

dorimedini-starkware reviewed (changes requested):
> Reviewed 1 of 2 files at r6, 3 of 3 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1340)* status: all files reviewed, 4 unresolved discussions (waiting on @ArielElp, @ilyalesokhin-starkware, and @zuphitf)

___
*[`crates/blockifier/src/execution/entry_point_execution.rs` line 119 at r1](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoGL2dh8e4Y55eCgNvc:-NofRigiEFMfy8FpJOTr:b-v4j6dt) ([raw file](https://github.com/starkware-libs/blockifier/blob/edfd949a2a21651399b413a5c24d1a0cbc74346b/crates/blockifier/src/execution/entry_point_execution.rs#L119)):*
<details><summary><i>Previously, ilyalesokhin-starkware wrote…</i></summary><blockquote>

By doing them in the same PR you make it hard to see the change.
</blockquote></details>

usually I would agree but IMO there is good separation between testing and feature in this PR; easy to switch contexts
___
*[`crates/blockifier/src/execution/entry_point_execution.rs` line 57 at r7](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NofR1im2QOOk3igQLl4:-NofR1im2QOOk3igQLl5:b-rtagwg) ([raw file](https://github.com/starkware-libs/blockifier/blob/3cde14762e8dfd62c13d7c48820be5761d996352/crates/blockifier/src/execution/entry_point_execution.rs#L57)):*
> ```XML
>     context: &mut EntryPointExecutionContext,
> ) -> EntryPointExecutionResult<CallInfo> {
>     // Fetch the address from `call`.
> ```

delete



_Code quote:_
```XML
// Fetch the address from `call`.
```
___
*[`crates/blockifier/src/execution/entry_point_test.rs` line 633 at r5](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoN2JD95eOzGCrLJdku:-NofROmz7CSTcMJajgQd:b-s5k13l) ([raw file](https://github.com/starkware-libs/blockifier/blob/a770b76e9a8491012e57c31f532216e3e009226f/crates/blockifier/src/execution/entry_point_test.rs#L633)):*
<details><summary><i>Previously, zuphitf wrote…</i></summary><blockquote>

Didn't we discuss with Ariel that this is in the todo / nice to have list for the future refactoring and not a requirement for this PR?
</blockquote></details>

we are not manipulating the strings yet:
1. first we put regression test in place (for both cairo versions)
2. fix immediate bugs
3. refactor stack trace handling and construction
4. reformat (once we have structured stack) to nice error text


<!-- Sent from Reviewable.io -->

zuphitf commented:
*[`crates/blockifier/src/execution/entry_point_execution.rs` line 57 at r7](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NofR1im2QOOk3igQLl4:-NofSZxfAhCN_nePO5Eh:b-896fix) ([raw file](https://github.com/starkware-libs/blockifier/blob/3cde14762e8dfd62c13d7c48820be5761d996352/crates/blockifier/src/execution/entry_point_execution.rs#L57)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

delete
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

dorimedini-starkware reviewed (commented):
> Reviewed 1 of 1 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1340)* status: all files reviewed, 3 unresolved discussions (waiting on @ArielElp and @ilyalesokhin-starkware)



<!-- Sent from Reviewable.io -->

ilyalesokhin-starkware commented:
*[`crates/blockifier/src/execution/entry_point_test.rs` line 633 at r5](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoN2JD95eOzGCrLJdku:-Nof_gkcCqmCTm1pHsJS:bd38kdn) ([raw file](https://github.com/starkware-libs/blockifier/blob/a770b76e9a8491012e57c31f532216e3e009226f/crates/blockifier/src/execution/entry_point_test.rs#L633)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

we are not manipulating the strings yet:
1. first we put regression test in place (for both cairo versions)
2. fix immediate bugs
3. refactor stack trace handling and construction
4. reformat (once we have structured stack) to nice error text
</blockquote></details>

What is included in this PR?
it seems that more then 1.


<!-- Sent from Reviewable.io -->

ilyalesokhin-starkware commented:
*[`crates/blockifier/src/execution/entry_point.rs` line 118 at r8](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-Nof_ykp2P6R_tfZLWth:-Nof_ykp2P6R_tfZLWti:b-foxb2j) ([raw file](https://github.com/starkware-libs/blockifier/blob/c1899118691f15634f1b3d0a2e1be97ad5b4281b/crates/blockifier/src/execution/entry_point.rs#L118)):*
> ```XML
>                 EntryPointExecutionError::ExecutionFailed { .. } => {
>                     context.error_stack.push((storage_address, format!("{}\n", &error)));
>                     error
> ```

Return a different error here.



_Code quote:_
```XML
error
```


<!-- Sent from Reviewable.io -->

ilyalesokhin-starkware commented:
*[`crates/blockifier/src/execution/entry_point_execution.rs` line 122 at r1](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoIZ-1_A2dMZhiUOamM:-Nofa4tq8O2AsCFn8r0l:boqncq0) ([raw file](https://github.com/starkware-libs/blockifier/blob/edfd949a2a21651399b413a5c24d1a0cbc74346b/crates/blockifier/src/execution/entry_point_execution.rs#L122)):*
<details><summary><i>Previously, zuphitf wrote…</i></summary><blockquote>

I don't think it's a good idea.
In fact, I don't think this is the right place to do this.
I'm reverting the changes in this file and adding the missing match in the wrapping function, where it belongs.
</blockquote></details>

Moved my comment to the new location...


<!-- Sent from Reviewable.io -->

dorimedini-starkware reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1340)* status: all files reviewed, 3 unresolved discussions (waiting on @ArielElp, @ilyalesokhin-starkware, and @zuphitf)

___
*[`crates/blockifier/src/execution/entry_point_test.rs` line 633 at r5](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoN2JD95eOzGCrLJdku:-NofiqQBDefHuN9iY4iw:b-9l38ns) ([raw file](https://github.com/starkware-libs/blockifier/blob/a770b76e9a8491012e57c31f532216e3e009226f/crates/blockifier/src/execution/entry_point_test.rs#L633)):*
<details><summary><i>Previously, ilyalesokhin-starkware wrote…</i></summary><blockquote>

What is included in this PR?
it seems that more then 1.
</blockquote></details>

1 was here for cairo0; I guess the correct thing to do would have been to do (1) first *with the bug* (so regression test would tes something bad), and then do (2).
since (2) was found before (1) was done this was combined.
@zuphitf how painful is it to split like this?
add regression test without bugfix, and then bugfix?


<!-- Sent from Reviewable.io -->

zuphitf commented:
*[`crates/blockifier/src/execution/entry_point_test.rs` line 633 at r5](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoN2JD95eOzGCrLJdku:-Nog6Cga7oTmPQiueLeu:bqlckgr) ([raw file](https://github.com/starkware-libs/blockifier/blob/a770b76e9a8491012e57c31f532216e3e009226f/crates/blockifier/src/execution/entry_point_test.rs#L633)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

1 was here for cairo0; I guess the correct thing to do would have been to do (1) first *with the bug* (so regression test would tes something bad), and then do (2).
since (2) was found before (1) was done this was combined.
@zuphitf how painful is it to split like this?
add regression test without bugfix, and then bugfix?
</blockquote></details>

It's doable, but why would we want that?
This isn't a big PR and it's a good practice to  introduce such changes together with a test, esp in this case where it's not at all clear what (and if) we're fixing without something like this regression test's output.


<!-- Sent from Reviewable.io -->

zuphitf commented:
*[`crates/blockifier/src/execution/entry_point_test.rs` line 633 at r5](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-NoN2JD95eOzGCrLJdku:-Nog6qMG7kvWRVCw6vKx:b5znd9g) ([raw file](https://github.com/starkware-libs/blockifier/blob/a770b76e9a8491012e57c31f532216e3e009226f/crates/blockifier/src/execution/entry_point_test.rs#L633)):*
<details><summary><i>Previously, zuphitf wrote…</i></summary><blockquote>

It's doable, but why would we want that?
This isn't a big PR and it's a good practice to  introduce such changes together with a test, esp in this case where it's not at all clear what (and if) we're fixing without something like this regression test's output.
</blockquote></details>

The changes in the test contracts that take most of the volume here are trivial


<!-- Sent from Reviewable.io -->

ilyalesokhin-starkware commented:
*[`crates/blockifier/src/execution/entry_point.rs` line 118 at r8](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-Nof_ykp2P6R_tfZLWth:-NogPWVc0P91qvO9oN5H:bo9iteg) ([raw file](https://github.com/starkware-libs/blockifier/blob/c1899118691f15634f1b3d0a2e1be97ad5b4281b/crates/blockifier/src/execution/entry_point.rs#L118)):*
<details><summary><i>Previously, ilyalesokhin-starkware wrote…</i></summary><blockquote>

Return a different error here.
</blockquote></details>

SyscallError { error_data: vec![]}, 


<!-- Sent from Reviewable.io -->

ilyalesokhin-starkware reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 3 of 8 files at r3, 3 of 3 files at r7, 1 of 1 files at r8, 1 of 1 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-:-NogXbZR-iQmXh26fpze:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @ArielElp)



<!-- Sent from Reviewable.io -->

dorimedini-starkware reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1340#-:-NogZ07O7NqGsMDizl6y:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @ArielElp)



<!-- Sent from Reviewable.io -->
